rules/static_framework_generator.bzl

```diff
@@ -0,0 +1,214 @@
+################
+
+def _declare_directory(ctx, framework_name, filename):
+    return ctx.actions.declare_directory("%s/%s/%s" % (ctx.label.name, framework_name, filename))
+
+def _declare_resource_bundles(ctx, target):
+    resource_bundles = []
+    for resource_target in _get_resource_targets([target]).to_list():
+        if AppleResourceInfo in resource_target:
+            for resource_group in resource_target[AppleResourceInfo].unprocessed:
+                resource_files = resource_group[2].to_list()
+                if len(resource_files) == 0:
+                    continue
+
+                bundle = _declare_directory(ctx, _get_existing_framework_name(target), "resources/%s" % resource_group[0])
+                paths_to_copy = " ".join([file.path for file in resource_files])
+                ctx.actions.run_shell(
+                    outputs = [bundle],
+                    inputs = resource_files,
+                    command = "ditto $1 $2",
+                    arguments = [paths_to_copy, bundle.path],
+                )
+                resource_bundles.append(bundle)
+    return resource_bundles
+
+def _get_existing_framework_path(target):
+    existing_framework_files = target.files.to_list()
+    if existing_framework_files:
+        framework_path = existing_framework_files[0].dirname
+        if framework_path.endswith(".framework"):
+            return framework_path
+
+    fail("The target %s does not produce any framework" % target.label.name)
+
+def _get_existing_framework_name(target):
+    existing_framework_files = target.files.to_list()
+    if existing_framework_files:
+        subpaths = existing_framework_files[0].dirname.split("/")
+        if subpaths:
+            framework_name = subpaths[-1]
+            if framework_name.endswith(".framework"):
+                return framework_name.replace(".framework", "")
+
+    fail("The target %s does not produce any framework" % target.label.name)
+
+def _declare_framework(ctx, target):
+    framework_name = _get_existing_framework_name(target)
+    return _declare_directory(ctx, framework_name, framework_name + ".framework")
+
+def _make_frameworks(ctx):
+    frameworks = []
+    for target in ctx.attr.targets:
+        framework = _declare_framework(ctx, target)
+        resource_bundles = _declare_resource_bundles(ctx, target)
+
+        existing_framework_files = target.files.to_list()
+        existing_framework_path = _get_existing_framework_path(target)
+
+        resource_paths = " ".join([bundle.path for bundle in resource_bundles])
+        ctx.actions.run_shell(
+            outputs = [framework],
+            inputs = existing_framework_files + resource_bundles,
+            command = 'rsync -a "$1/" "$3"; for resource in $2; do rsync -a "$resource" "$3/Resources"; done',
+            arguments = [existing_framework_path, resource_paths, framework.path],
+        )
+        frameworks.append(framework)
+    return frameworks
+
+def _make_destination(ctx):
```

Comment:
Perhaps check if the `destination` is relative VS absolute?

Reply:
Ill try to rework this to place the outputs either in bazel-out or bazel-bin